### ISSUE TITLE
Feature request: Run on other branches

### ISSUE THREAD

**Issue by paulnovo** [CodeDocs/CodeDocs#4]
Currently codedocs runs on the default github branch (usually master). Add the ability to set other branches to run on.

One idea is to run on any branch that contains a .codedocs file.

**Comment by MarkCallow** [comment]
I'd prefer to explicity say which branches with lines in the .codedocs file. This is because you can easily end up with conflicts in the .codedocs file every time you merge any code between branches if they differ.

**Comment by MarkCallow** [comment]
@LB-- what is confusing? You suggested running on any branch with a .codedocs file. I said that I want to have the same .codedocs file in every branch to avoid merge issues. I don't necessarily want to build documentation in all those branches. Hence I requested a way to explicitly state in the .codedocs file which branches to build, similar to the `branches` directive in a Travis-CI .yml file.

**Comment by LB--** [comment]
@msc- I don't understand the correlation between including the list of branches in the file and merge conflicts. IMO metadata like this shouldn't be in the repository at all - I'm not a fan of the way travis does this. The option for what/which branch(es) to build should be on the CodeDocs web interface. As for merge conflicts, git affords you many ways to resolve them including just manually making the file the way you want it to be - really this shouldn't even be a deciding factor considering how trivial it is. Just my personal preference though.

**Comment by MarkCallow** [comment]
@paulnovo Any progress on this feature?

@LB I tried to have a file before that permanently differed between my incoming and master branches. I thought that the first time I did an incoming->master merge and resolved the conflict that would be it. Done. However every time I did a merge subsequently I got a conflict on this file even though there had  been no further changes to it in either branch. So I gave up. Perhaps I wasn't using Git correctly.

Having the option for which branches to build on the CodeDocs web interface would be fine. I agree that is better than a branches specifier in .codedocs.

**Comment by LB--** [comment]
@msc-

> I tried to have a file before that permanently differed between my incoming and master branches.

Oh, this wasn't what I assumed you meant. Since the only conflicts we are talking about here would be the addition or removal of branch names in the .codedocs file, I'm not really sure what the issue is. I think travis for example only reads which branches to build from the main branch and then uses the actual instructions from the branch being built. So, you can easily add the new branch in a commit to master, then actually create the branch and make any needed changes and later merge it back into master. Finally when you want to remove the branch, just do so and create another commit on master to remove it from the list. I don't imagine many merge conflicts coming from this.

_However_, as I said (and as you agreed) I'd still prefer it if the list of branches to build _was not_ stored in any file inside the repository - that way you don't have to jump through any hoops as detailed above.

**Comment by paulnovo** [comment]
@LB-- @msc I agree, doing it in the web interface makes the most sense to me. This will require some significant changes to get up and running. I am hoping to keep the default branches documentation under `https://codedocs.xyz/[User]/[Repo]/` while maybe a branch is at `https://codedocs.xyz/[User]/[Repo]/[Branch]/`. What do you think about that?

**Comment by MarkCallow** [comment]
Any progress? I see in [openscenegraph issue 164](https://github.com/openscenegraph/OpenSceneGraph/issues/164) that you said last November you were working on it.

I could really use this right now. I have a branch of my project with major API additions that I would like people to review before I merge to master. The documentation is the best way for reviewers to do that.

**Comment by MarkCallow** [comment]
Ping. Again!

> I am hoping to keep the default branches documentation under https://codedocs.xyz/[User]/[Repo]/ while maybe a branch is at https://codedocs.xyz/[User]/[Repo]/[Branch]/.

This is fine. It would probably be good also if https://codedocs.xyz/[User]/[Repo]/[Default] redirected to https://codedocs.xyz/[User]/[Repo] or vice versa.

**Comment by MarkCallow** [comment]
Another year passes. No activity. Is codedocs dead?

I am now building the docs for my other branch on Travis-CI and deploying them to gh-pages. I will do that for the master too in the near future because of this and issue #22.

**Comment by eyalroz** [comment]
I'm also anxious to see this one implemented, but it seems from @MarkCallow 's pings here that this is not going anywhere...

**Comment by neel** [comment]
I see this is an old issue. Any progress so far ?

**Comment by ABRG-Models** [comment]
This is a black lives matter issue. I have switched the name of a repository's 'master' branch to 'main' to remove the slavery connotations of our software. Now it doesn't work with codedocs, which is a shame.
